test/multiple-user-principal-types.test.js

```diff
@@ -393,37 +395,194 @@ describe('Multiple users with custom principalType', function() {
-              type: 'belongsTo',
-              model: 'OneUser',
-              foreignKey: 'userId',
-            },
-          },
         });
-        app.model(Album, {dataSource: 'db'});
 
-        return Album.create({name: 'album', userId: userFromOneModel.id})
+        it('legacy behavior resolves false without belongsTo relation', function() {
+          var Album = app.registry.createModel('Album', {
+            name: String,
+            userId: Number,
+            owner: Number,
+          });
+          app.model(Album, {dataSource: 'db'});
+
+          return Album.create({
+            name: 'album',
+            userId: userFromOneModel.id,
+            owner: userFromOneModel.id,
+          })
           .then(function(album) {
-            var invalidContext = {
-              principalType: AnotherUser.modelName,
+            var validContext = {
+              principalType: OneUser.modelName,
               principalId: userFromOneModel.id,
               model: Album,
               id: album.id,
             };
-            return Role.isInRole(Role.OWNER, invalidContext);
+            return Role.isInRole(Role.OWNER, validContext);
           })
           .then(function(isOwner) {
             expect(isOwner).to.be.false();
           });
+        });
+
+        it('legacy behavior resolves false if owner has incorrect principalType', function() {
+          var Album = app.registry.createModel('Album', {
+            name: String,
+            userId: Number,
+          }, {
+            relations: {
+              user: {
+                type: 'belongsTo',
+                model: 'OneUser',
+                foreignKey: 'userId',
+              },
+            },
+          });
+          app.model(Album, {dataSource: 'db'});
+
+          return Album.create({name: 'album', userId: userFromOneModel.id})
+          .then(function(album) {
+            var invalidPrincipalTypes = [
+              'invalidContextName',
+              'USER',
+              AnotherUser.modelName,
+            ];
+            var invalidContexts = invalidPrincipalTypes.map(principalType => {
+              return {
+                principalType,
+                principalId: userFromOneModel.id,
+                model: Album,
+                id: album.id,
+              };
+            });
+            return Promise.map(invalidContexts, context => {
+              return Role.isInRole(Role.OWNER, context)
+                .then(isOwner => {
+                  return {
+                    principalType: context.principalType,
+                    isOwner,
+                  };
+                });
+            });
+          })
+          .then(result => {
+            expect(result).to.eql([
+              {principalType: 'invalidContextName', isOwner: false},
+              {principalType: 'USER', isOwner: false},
+              {principalType: AnotherUser.modelName, isOwner: false},
+            ]);
+          });
+        });
+
```

Comment:
refactored 3 tests in one to check for various incorrect principalType while keeping the error messages explicit